units/systemd-quotacheck-root.service.in

```diff
@@ -0,0 +1,24 @@
+#  SPDX-License-Identifier: LGPL-2.1-or-later
+#
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=File System Quota Check
+Documentation=man:systemd-quotacheck.service(8)
+
+ConditionPathExists={{QUOTACHECK}}
+
+DefaultDependencies=no
+After=initrd-switch-root.target
+Before=local-fs.target
+
```

Comment:
does the rootfs need to be writable when quotacheck runs? i'd assume so, no? if so, you also need After=systemd-remount-fs.service which ensures the command is only run once the root fs has been remounted writable

Reply:
i mean, the template service has the dep, so this one almost certainly should too?

Reply:
or actually, the template service really shouldn't have it, there's no reason for waiting for systemd-remount-fs.service for them, because non-root fs are already mounted writable right-away

Reply:
Yes, quotacheck needs the rootfs rw, for adding/modifying *quota.user *quota.group files.
The template should probably be ordered after the mount.
-->fixed